modules/material/material.service.js

```diff
@@ -32,11 +36,46 @@ class MaterialsService {
     return await Material.findByIdAndUpdate(id, material, { new: true });
   }
 
-  async addMaterial(data) {
-    if (await this.checkMaterialExist(data)) {
+  async addMaterial({ material, images }) {
+    const { additionalPrice, ...rest } = material;
+
+    const currency = await Currency.findOne();
+
+    if (await this.checkMaterialExist(rest)) {
```

Comment:
Function 'checkMaterialExist' require 2 arguments.

Reply:
second argument is optional